### PR TITLE
Fixed a bug where 'enabled' on envs and systems broke cibyl

### PR DESCRIPTION
**Pull request by jsanemet**
- Added 'enabled' as a property of env, system and zuul.d on JSON schema.
- Make cibyl not interpret 'enabled' as another system name.
- Enabling the test case that saw this happen.

### CODE REVIEW

File: tests/cibyl/e2e/test_config.py [hunk 1]
```diff
@@ -45,7 +45,6 @@ class TestSchema(EndToEndTest):
     """Tests configuration files against their schema.
     """
 
-    @skip(reason='Test fails to a reason different to what is tested here')
```
Comment (cescgina) [inline tests/cibyl/e2e/test_config.py:48]: skip is removed but still imported, flake is complaining in CI about it

Comment (jsanemet) [reply tests/cibyl/e2e/test_config.py:48]: True

Comment (jsanemet) [reply tests/cibyl/e2e/test_config.py:48]: Fixed

**Review by cescgina — APPROVED** [review]
Good catch, +1

**Pull request merged**